plugins/uefi/fu-uefi-common.c

```diff
@@ -255,10 +256,23 @@ fu_uefi_get_esp_path_for_os (const gchar *esp_path)
 	}
 	if (os_release_id == NULL)
 		os_release_id = "unknown";
+	esp_path = g_build_filename (base, "EFI", os_release_id, NULL);
+	if (!g_file_test (esp_path, G_FILE_TEST_IS_DIR)) {
+		const gchar *id_like_id = g_hash_table_lookup (os_release, "ID_LIKE");
+		if (id_like_id != NULL) {
+			g_autofree gchar* id_like_path = g_build_filename (base, "EFI", id_like_id, NULL);
+			if (g_file_test (esp_path, G_FILE_TEST_IS_DIR)) {
+				free (esp_path);
```

Comment:
`g_free`

Reply:
This all seems a bit clumsy tbh..

Reply:
Yeah you're right.  I  think it was leaking memory too.  I just reworked it some.